src/pyicloud_ipd/base.py

```diff
@@ -190,30 +193,81 @@ def authenticate(self, force_refresh:bool=False, service:Optional[Any]=None) ->
+                def __init__(self, password: str):
+                    self.pwd = password
+
+                def set_encrypt_info(self, salt: bytes, iterations: int):
+                    self.salt = salt
+                    self.iterations = iterations
+
+                def encode(self):
+                    key_length = 32
+                    return hashlib.pbkdf2_hmac('sha256', hashlib.sha256(self.pwd.encode()).digest(), self.salt, self.iterations, key_length)
+
+            # Step 1: client generates private key a (stored in srp.User) and public key A, sends to server
+            srp_password = SrpPassword(self.user["password"])
+            srp.rfc5054_enable()
+            srp.no_username_in_x()
+            usr = srp.User(self.user["accountName"], srp_password, hash_alg=srp.SHA256)
+            uname, A = usr.start_authentication()
+            data = {
+                'a': base64.b64encode(A).decode(),
+                'accountName': uname,
+                'protocols': ['s2k', 's2k_fo']
+            }
+
+            try:
+                response = self.session.post("%s/signin/init" % self.AUTH_ENDPOINT, data=json.dumps(data), headers=headers)
+                if response.status_code == 401:
+                    raise PyiCloudAPIResponseException(response.status_code)
+            except PyiCloudAPIResponseException as error:
+                msg = "Failed to initiate srp authentication."
+                raise PyiCloudFailedLoginException(msg, error) from error
+
+            # Step 2: server sends public key B, salt, and c to client
+            body = response.json()
+            salt = base64.b64decode(body['salt'])
+            b = base64.b64decode(body['b'])
+            c = body['c']
+            iterations = body['iteration']
+
+            # Step 3: client generates session key M1 and M2 with salt and b, sends to server
+            srp_password.set_encrypt_info(salt, iterations)
+            m1 = usr.process_challenge( salt, b )
+            m2 = usr.H_AMK
+
+            data = {
+                "accountName": uname,
+                "c": c,
+                "m1": base64.b64encode(m1).decode(),
+                "m2": base64.b64encode(m2).decode(),
+                "rememberMe": True,
+                "trustTokens": [],
+            }
+
+            if self.session_data.get("trust_token"):
+                data["trustTokens"] = [self.session_data.get("trust_token")]
+
             try:
-                self.session.post(
-                    "%s/signin" % self.AUTH_ENDPOINT,
+                response = self.session.post(
+                    "%s/signin/complete" % self.AUTH_ENDPOINT,
                     params={"isRememberMeEnabled": "true"},
                     data=json.dumps(data),
                     headers=headers,
                 )
+                if response.status_code == 401:
+                    raise PyiCloudAPIResponseException(response.status_code)
```

Comment:
```suggestion
                    raise PyiCloudAPIResponseException(response.message , str(response.status_code))
```

`PyiCloudAPIResponseException` takes two [parameters](https://github.com/icloud-photos-downloader/icloud_photos_downloader/blob/8b3adb04eb13cf47c5684530348c7ee347c85fcc/src/pyicloud_ipd/exceptions.py#L13), both must be strings

Reply:
Fixed, thank you.

Reply:
Thank you for getting this fixed.